slack-lint-checks/src/main/java/slack/lint/rx/RxObservableEmitDetector.kt

```diff
@@ -0,0 +1,99 @@
+// Copyright (C) 2025 Slack Technologies, LLC
+// SPDX-License-Identifier: Apache-2.0
+package slack.lint.rx
+
+import com.android.tools.lint.checks.DataFlowAnalyzer
+import com.android.tools.lint.client.api.UElementHandler
+import com.android.tools.lint.detector.api.Category
+import com.android.tools.lint.detector.api.Detector
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.JavaContext
+import com.android.tools.lint.detector.api.Severity
+import com.android.tools.lint.detector.api.SourceCodeScanner
+import org.jetbrains.uast.UCallExpression
+import org.jetbrains.uast.ULambdaExpression
+import slack.lint.util.sourceImplementation
+
+class RxObservableEmitDetector : Detector(), SourceCodeScanner {
+  override fun getApplicableUastTypes() = listOf(UCallExpression::class.java)
+
+  override fun createUastHandler(context: JavaContext): UElementHandler {
+    return object : UElementHandler() {
+      override fun visitCallExpression(node: UCallExpression) {
+        val issue = functionToIssue[node.methodName] ?: return
+        val lambdaExpression = node.valueArguments.lastOrNull() as? ULambdaExpression ?: return
+        val producerScopeParam = lambdaExpression.parameters.firstOrNull() ?: return
+
+        // Verify the parameter is a ProducerScope
+        if (!producerScopeParam.type.canonicalText.startsWith(PROVIDER_SCOPE_FQN)) return
```

Comment:
this stringy check is gonna be brittle, can you not resolve the qualified name of the `type` and compare it directly?

Reply:
Agreed. I could not find a better way. Any guidance on how to do this would be great

Reply:
```kotlin
val paramType = producerScopeParam.type
if (paramType !is PsiClassType) return
if (paramType.resolve()?.qualifiedName != PROVIDER_SCOPE_FQN) return
```

Reply:
https://github.com/slackhq/slack-lints/pull/375/commits/3479bf6ae021ff3ad2a26148fdf02aa5ac9abf69